libsolidity/lsp/Transport.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#include <libsolidity/lsp/Transport.h>
+
+#include <libsolutil/JSON.h>
+#include <libsolutil/Visitor.h>
+#include <libsolutil/CommonIO.h>
+
+#include <boost/algorithm/string.hpp>
+
+#include <iostream>
+#include <sstream>
+
+using namespace std;
+using namespace solidity::lsp;
+
+JSONTransport::JSONTransport(istream& _in, ostream& _out):
+	m_input{_in},
+	m_output{_out}
+{
+}
+
+JSONTransport::JSONTransport(): JSONTransport(cin, cout)
```

Comment:
I'd stick to either putting it on same line in both constructors or in neither. Looking at the previous constructor and the empty body here, for a moment I was confused why there's no initialization :)
```suggestion
JSONTransport::JSONTransport():
	JSONTransport(cin, cout)
```